lib/Net/GitHub/V3/Query.pm

```diff
@@ -230,7 +230,7 @@ sub query {
             my $uri = URI->new($url);
             my %query_form = $uri->query_form;
             $query_form{per_page} ||= $self->per_page;
-            $uri->query_form(%query_form);
+            $uri->query_form(%query_form, ref $data eq 'HASH' ? %$data : () );
```

Comment:
This is for `GET` request like `get_content`
I can probably try to limit this line to `GET` only, as I agree with you for other requests it will be in the JSON

Reply:
note that a few lines above there is a check for `GET` so this is only for `GET`
I do not think it's an issue
line 227: `if ($request_method eq 'GET') {`
https://github.com/fayland/perl-net-github/pull/95/files#diff-ca2efc3207169a073b8c029c07c90083L227

Reply:
hi, if there is no obvious reason to have $data into url, I'd like to rollback the changes. Thanks

Reply:
This is needed for `get_content` in order to request a specific `ref` state in the `GET` url.
If you revert it, then it will not be possible to request a specific state of a file

Reply:
could you show an example? probably it can be handled with %query_form?

Reply:
Here is for example an example usage to get a file from a specific branch or commit
```perl
my $api_answer = $gh->repos->get_content(
            {   owner => 'atoomic', repo => 'perl-net-github',
                path  => 'lib/Net/GitHub/V3/Query.pm'
            },
            { ref => 'named-parameters' }
);
```
without this we cannot pass the `ref` argument to request for a specific state